src/main/scala/fs2/Process1.scala

```diff
@@ -45,6 +45,29 @@ object process1 {
         case chunk #: h => Pull.output(chunk) >> take(n - chunk.size.toLong).apply(h)
       }
 
+  def takeWhile[F[_], I](p: I => Boolean)(implicit F: NotNothing[F]): Handle[F, I] => Pull[F, I, Handle[F, I]] = 
+    h => 
+      Pull.await1Option(h).flatMap { 
+        case Some(c #: h) if p(c) =>
+          Pull.output1(c) >> takeWhile(p).apply(h)
+        case _ => Pull.done
+      }
+      
+  def drop[F[_], I](n: Long)(implicit F: NotNothing[F]): Handle[F, I] => Pull[F, I, Handle[F, I]] = 
+    h =>
+      if (n <= 0) id.apply(h)
+      else Pull.awaitLimit(if (n <= Int.MaxValue) n.toInt else Int.MaxValue)(h).flatMap {
+        case chunk #: h => drop(n - chunk.size).apply(h)
+      }
+      
+  def dropWhile[F[_], I](p: I => Boolean)(implicit F: NotNothing[F]): Handle[F, I] => Pull[F, I, Handle[F, I]] =
+    h => 
+      Pull.await1Option(h).flatMap {
```

Comment:
@pchiusano I think that makes sense to me too. I will have to update the PR after hours though, since firm proxy won't allow me to push out code. If by then the method is still not added, I will add it myself.